### PR TITLE
Upstream local changes for supporting Slashdot and Bleacher Report RSS links

### PR DESCRIPTION
**Pull request by j-roskopf**
Add support for image links inside link tag on articles (as seen on bleacher report https://bleacherreport.com/articles/feed)

Screenshot 2023-05-28 at 8 49 39 PM

Add support for categories defined in attribute 'term' (as seen on slashdot http://rss.slashdot.org/Slashdot/slashdotMainatom)

Screenshot 2023-05-28 at 8 50 14 PM

All of the other things I have done were addressed with the initial + additional Atom RSS PRs (https://github.com/prof18/RSS-Parser/pull/94, https://github.com/prof18/RSS-Parser/pull/114)

Please let me know if there is anything else you'd like to see done with this pull request!

### CODE REVIEW

File: rssparser/src/main/java/com/prof/rssparser/core/parser/rss/RssParser.kt [hunk 1]
```diff
@@ -263,7 +264,19 @@ internal fun extractRSSContent(xmlPullParser: XmlPullParser): Channel {
...
-                    insideItem -> articleBuilder.image(xmlPullParser.nextTrimmedText())
+                    insideItem -> {
+                        try {
+                            // It would be ideal if we could peek ahead to see what is coming.
+                            // Attempt to get the image text if it's not contained in another tag
+                            articleBuilder.image(xmlPullParser.nextTrimmedText())
+                        } catch (exception: XmlPullParserException) {
+                            // if we get an exception from grabbing the next text,
+                            // assume we are inside the image tag and look inside the corresponding link tag and consume that
+                            if(xmlPullParser.contains(RSSKeyword.Link)) {
```
Comment (j-roskopf) [inline rssparser/src/main/java/com/prof/rssparser/core/parser/rss/RssParser.kt:275]: definitely am not in love with this strategy - definitely open to other solutions!

**Review by prof18 — CHANGES_REQUESTED** [review]

File: rssparser/src/main/java/com/prof/rssparser/core/parser/rss/RssParser.kt (hunk 1 above)
Comment (j-roskopf) [reply rssparser/src/main/java/com/prof/rssparser/core/parser/rss/RssParser.kt:275]: hmm, maybe I'm not following entirely but I believe the link provided covers a different scenario than what this is dealing with. There aren't any attributes to grab on the image or link tag in this example to probe. 

In the scenario above, the RSS we want to parse looks something like

```
<image>      <link>https://media.bleacherreport.com/image/upload/x_0,y_43,w_1800,h_1200,c_crop/v1685197298/ieh5bpyilj43f4dzpgua.jpg</link>
</image>
```

I believe that would work if the RSS looked like

```
<image link="https://media.bleacherreport.com/image/upload/x_0,y_43,w_1800,h_1200,c_crop/v1685197298/ieh5bpyilj43f4dzpgua.jpg"/>
```

Please let me know if I'm misunderstanding the approach you suggested!

Comment (prof18) [reply rssparser/src/main/java/com/prof/rssparser/core/parser/rss/RssParser.kt:275]: Ah right 🤔 

Alright, let's keep this logic for now and I will check if the Parser API provides a cleaner way.

**Review by prof18 — APPROVED** [review]
Thanks a lot for the contribution! 🚀 
Really appreciated! 🙏

File: rssparser/src/main/java/com/prof/rssparser/core/parser/rss/RssParser.kt (hunk 1 above)
Comment (prof18) [reply rssparser/src/main/java/com/prof/rssparser/core/parser/rss/RssParser.kt:275]: This seems to work! 

```kotlin
  xmlPullParser.next()
  val text = xmlPullParser.text.trim()
  // Get the image text if it's not contained in another tag
  if (text.isNotEmpty()) {
      articleBuilder.image(text)
  } else {
      xmlPullParser.next()
      if(xmlPullParser.contains(RSSKeyword.Link)) {
          articleBuilder.image(xmlPullParser.nextTrimmedText())
      }
  }
```

Weirdly, the call to `xmlPullParser.next()` always return the `EventType` `TEXT` even though there's a `link`. But if there's a `link` the current text is empty.

**Pull request merged**